scripts/truffle/bot.js

```diff
@@ -70,47 +81,81 @@ module.exports = async function (callback) {
 
 async function processCdps() {
   let maxCdp = await bCdpManager.cdpi();
-  for (let i = 1; i <= maxCdp; i++) {
-    const isPending = pending.get(i);
-    if (!isPending) {
-      await processCdp(i);
+  for (let cdp = 1; cdp <= maxCdp; cdp++) {
+    if (!pending.get(cdp)) {
+      pending.set(cdp, true);
+      try {
+        await processCdp(cdp);
+      } catch (err) {
+        console.log(err);
+      }
+      pending.set(cdp, false);
     }
   }
 }
 
 async function processCdp(cdp) {
-  pending.set(cdp, true);
+  try {
+    const cdpInfo = await liqInfo.getCdpData.call(
+      cdp,
+      cdp,
+      MEMBER_1,
+      await getEth2DaiMarketPrice(),
```

Comment:
here it should be the meadnizer, and ideally should be read only once, outside the for loop.